agent/src/agentclient.rs

```diff
@@ -182,15 +183,14 @@ impl<T: APIServerClient> BrupopAgent<T> {
     /// create the custom resource associated with this node
     pub async fn create_metadata_custom_resource(&mut self) -> Result<()> {
```

Comment:
Something that I've grown quite fond of us `tracing::instrument`. It's a macro that you can associate with methods in order to automatically generate a span every time the function is called. The macro also automatically emits an `ERROR` level message if the function returns an `Err`. Pretty handy.

I'd suggest replacing the `event!` in this method with an `#[instrument(skip(self))]` on the method call.